README.md

```diff
@@ -1,26 +1,212 @@
+[`trs_filer/config.yaml`][trs-filer-config]. As the format follows the one
+expected by [FOCA][res-foca], refer to the FOCA documentation for a description
+of the various configuration options.
+
+The only exception to this is the custom section `endpoints`, which lists all
+the configuration options that are specific to TRS-Filer. These are:
+
+* `service`: The parameters in this section are used to construct self
+  references/anchors as required by the [TRS specification][ga4gh-trs] and need
+  to be replaced with the corresponding values of each individual TRS-Filer
+  deployment.
+* `service-info`: The parameters in this section are used to populate the
+  service info database collection if such a collection does not yet exist,
+  usually when a new app instance is deployed. The default values should be
+  replaced for each individual deployment, in accordance with the schemas and
+  documentation provided in the [GA4GH Service Info
+  specification][ga4gh-service-info].
+* `tool`, `version` & `toolclass`: The parameters in these sections define the
+  behavior for generating the `id` and `meta_version` identifiers for tools,
+  tool versions and tool classes, respectively, as required by the [TRS
+  specification][ga4gh-trs].
+  * For `id`-type identifiers, random strings of a specified `length` are
+    generated from a set of allowed characters (`charset`). The latter option
+    accepts either a string, indicating the allowed characters, or a Python
+    expression that evaluates to such a string.
+  * For `meta_version`-type identifiers, increasing natural numbers are
+    generated that start with the value of `init` and are increased by
+    `increment` for each new resource.
+
+## Extension
+
+It is easy to add additional endpoints or modify the behavior of existing ones.
+
+All OpenAPI specifications are available in [`trs_filer.api`][trs-filer-api].
+Custom endpoints, i.e., those that are not specified in either the [GA4GH
+TRS][ga4gh-trs] or the [GA4GH Service Info][ga4gh-service-info] specifications,
+are included in file [`additions.openapi.yaml`][trs-filer-api-custom].
+
+When specifying new endpoints, the `operationId` directive indicates the name
+of the corresponding controller. By default, all controllers need to be
+implemented in the [`ga4gh.trs.server`][trs-filer-controllers] module. For
+example, when specifying an `operationId` of `myController` for a new endpoint
+in [`additions.openapi.yaml`][trs-filer-api-custom], the application would
+expect the corresponding controller to be implemented in function/class
+`ga4gh.trs.server.myController()`.
+
+> If necessary, the default location of the module containing the controllers
+> can be customized following instructions provided by [FOCA][res-foca],
+> [Connexion][res-connexion] and [OpenAPI][res-openapi].
+
+Custom errors can be specified in module
+[`errors.exceptions`][trs-filer-exceptions]. Just follow the existing structure
```

Comment:
Link to `errors.exceptions` is broken, link target needs to contain `trs_filer/` in path